src/panels/lovelace/cards/hui-grid-card.ts

```diff
@@ -0,0 +1,65 @@
+import { css, CSSResult } from "lit-element";
+import { computeCardSize } from "../common/compute-card-size";
+import { HuiStackCard } from "./hui-stack-card";
+import { GridCardConfig } from "./types";
+
+const DEFAULT_COLUMNS = 3;
+
+class HuiGridCard extends HuiStackCard<GridCardConfig> {
+  public async getCardSize(): Promise<number> {
+    if (!this._cards || !this._config) {
+      return 0;
+    }
+
+    const promises: Array<Promise<number> | number> = [];
+
+    for (const element of this._cards) {
+      promises.push(computeCardSize(element));
+    }
+
+    const results = await Promise.all(promises);
+
+    const maxCardSize = Math.max(...results);
+
+    return (
+      maxCardSize *
+      (this._cards.length / (this._config.columns || DEFAULT_COLUMNS))
+    );
+  }
+
+  setConfig(config: GridCardConfig) {
+    super.setConfig(config);
+    if ("columns" in config) {
+      this.style.setProperty(
+        "--grid-card-column-count",
+        String(config.columns)
+      );
+    } else {
+      this.style.removeProperty("--grid-card-column-count");
+    }
+  }
+
+  static get styles(): CSSResult[] {
+    return [
+      super.sharedStyles,
+      css`
+        #root {
+          display: grid;
+          grid-template-columns: repeat(
+            var(--grid-card-column-count, ${DEFAULT_COLUMNS}),
+            minmax(0, 1fr)
+          );
+          grid-gap: var(--grid-card-gap, 8px);
```

Comment:
`--grid-card-column-count` is defined if `columns` is added to config, otherwise removed and it will take whatever is used.

`--grid-card-gap` cannot be defined from the config, relies on users setting their own CSS vars.

Reply:
Why not just hard code the number in there. The JS in the Static CSS I though was no good?

Reply:
also, then the variable can be overwritten in a theme which is not something we want right?

Reply:
Why wouldn't we want a theme to change it? It's up to the user to decide to do so.

Reply:
We don't want a theme to change it if the card can. Doesn't seem like the correct way having two ways that can change it

Reply:
The gap I understand. I think that the grid column config should only be apart of the card config